app/crawler/naver_map_crawler.py

```
@@ -127,6 +151,38 @@ def _extract_apollo_state_sync(self, page) -> List[Dict]:
+    def _get_random_ua(self) -> str:
+        """Generates a random User-Agent string."""
+        fallback_ua = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
+        
+        if UserAgent is None:
+            logger.warning("fake-useragent not installed. Using fallback UA.")
+            return fallback_ua
+
+        try:
+            # NOTE: Added fallback argument to handle FakeUserAgentError robustly.
+            ua = UserAgent(browsers=['chrome', 'edge'], fallback=fallback_ua)
+            return ua.random
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)